### ISSUE TITLE
Consider switching to `libgoogle-cloud-storage` from `google-cloud-cpp`.

### ISSUE THREAD

**Issue by coryan** [conda-forge/arrow-cpp-feedstock#1305]
### Comment:

We have split `google-cloud-cpp` (https://github.com/conda-forge/google-cloud-cpp-feedstock/issues/141) this package can depend on a much smaller dependency:

https://anaconda.org/conda-forge/libgoogle-cloud-storage

and 

https://anaconda.org/conda-forge/libgoogle-cloud-storage-devel

Should be all it needs.

**Comment by coryan** [comment]
I am testing a PR locally. Will try to send it later today (New York time).

**Comment by coryan** [comment]
I must be doing something wrong, because my [changes](https://github.com/conda-forge/arrow-cpp-feedstock/compare/main...coryan:arrow-cpp-feedstock:use-smaller-package-for-google-cloud-storage) do not work:

```
Packaging libarrow-gandiva
[34mReloading output folder: [0m[34m/home/conda/feedstock_root/[0m[34mbuild_artifacts[0m
[?25l[2K[0G[?25h[?25l[2K[0Gfile:///home/conda/feedstock_root/build_artifact.. Checked  0.0s
file:///home/conda/feedstock_root/build_artifact.. Checked  0.0s
[?25h[?25l[2K[0G[+] 0.0s
file:///home/conda..  ⣾  [2K[1A[2K[0Gfile:///home/conda/feedstock_root/build_artifact..
file:///home/conda/feedstock_root/build_artifact..  96.0 B @   3.3MB/s  0.0s
[?25hMamba failed to solve:
 - _libgcc_mutex 0.1.* conda_forge
 - ca-certificates 2024.2.2.* hbcca054_0
 - libstdcxx-ng 13.2.0.* h7e041cc_5
 - libgomp 13.2.0.* h807b86a_5
 - _openmp_mutex 4.5.* 2_gnu
 - libgcc-ng 13.2.0.* h807b86a_5
 - icu 73.2.* h59595ed_0
 - libabseil 20240116.1.* cxx17_h59595ed_0
 - libiconv 1.17.* hd590300_2
 - libutf8proc 2.8.0.* h166bdaf_0
 - libzlib 1.2.13.* hd590300_5
 - openssl 3.2.1.* hd590300_0
 - xz 5.2.6.* h166bdaf_0
 - libre2-11 2023.06.02.* h5a48ba9_1
 - libxml2 2.12.5.* h232c23b_0
 - zstd 1.5.5.* hfc55251_0
 - libllvm15 15.0.7.* hb3ce162_4
 - re2 2023.06.02.* he738c6c_1
 - llvm 15.0.7.* hda56bd4_4
 - libarrow 15.0.0.* hf11b704_2_cpu

with channels:

The reported errors are:
- Encountered problems while solving:
-   - package libarrow-15.0.0-hf11b704_2_cpu requires libabseil >=20230802.1,<20230803.0a0, but none of the providers can be installed
- 

```

**Comment by h-vetinari** [comment]
There was an issue that's being fixed in #1306, after that it should work fine. I also just merged the migration PR, so you can just copy `libgoogle_cloud217.yaml` from the global pinning repo (resp. [here](https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/5496/files)) under `.ci_support/migrations` in this feedstock, and then a rerender should correctly populate the config YAMLs for the separate jobs.

**Comment by h-vetinari** [comment]
I had an excuse to try this, see #1314
